maven_test.go

```diff
@@ -162,3 +155,53 @@ func createHomeConfigAndLocalRepo(t *testing.T, encryptPassword bool) (err error
 	localRepoDir, err = ioutil.TempDir(os.Getenv(coreutils.HomeDir), "tmp.m2")
 	return err
 }
+
+func TestMavenBuildIncludePatterns(t *testing.T) {
+	initMavenTest(t, false)
+	buildNumber := "123"
+	commandArgs := []string{"--build-name=" + tests.MvnBuildName, "--build-number=" + buildNumber}
+	runMavenCleanInstall(t, createMultiMavenProject, tests.MavenIncludeExcludePatternsConfig, commandArgs)
+	assert.NoError(t, artifactoryCli.Exec("build-publish", tests.MvnBuildName, buildNumber))
+
+	// Validate deployed artifacts.
+	searchSpec, err := tests.CreateSpec(tests.SearchAllMaven)
+	assert.NoError(t, err)
+	verifyExistInArtifactory(tests.GetMavenMultiIncludedDeployedArtifacts(), searchSpec, t)
+
+	// Validate build info.
+	publishedBuildInfo, found, err := tests.GetBuildInfo(serverDetails, tests.MvnBuildName, buildNumber)
+	if err != nil {
+		assert.NoError(t, err)
+		return
+	}
+	if !found {
+		assert.True(t, found, "build info was expected to be found")
+		return
+	}
+	buildInfo := publishedBuildInfo.BuildInfo
+	if len(buildInfo.Modules) != 4 {
+		assert.Len(t, buildInfo.Modules, 4)
+		return
+	}
+	validateSpecificModule(buildInfo, t, 13, 2, 1, "org.jfrog.test:multi1:3.7-SNAPSHOT", buildinfo.Maven)
+	validateSpecificModule(buildInfo, t, 1, 0, 2, "org.jfrog.test:multi2:3.7-SNAPSHOT", buildinfo.Maven)
+	validateSpecificModule(buildInfo, t, 15, 1, 1, "org.jfrog.test:multi3:3.7-SNAPSHOT", buildinfo.Maven)
+	validateSpecificModule(buildInfo, t, 0, 1, 0, "org.jfrog.test:multi:3.7-SNAPSHOT", buildinfo.Maven)
+	cleanMavenTest()
+}
+
+func runMavenCleanInstall(t *testing.T, createProjectFunction func(*testing.T) string, configFileName string, additionalArgs []string) {
+	projDir := createProjectFunction(t)
+	configFilePath := filepath.Join(filepath.FromSlash(tests.GetTestResourcesPath()), "buildspecs", configFileName)
+	destPath := filepath.Join(projDir, ".jfrog", "projects")
+	createConfigFile(destPath, configFilePath, t)
+	assert.NoError(t, os.Rename(filepath.Join(destPath, configFileName), filepath.Join(destPath, "maven.yaml")))
+	oldHomeDir := changeWD(t, projDir)
+	repoLocalSystemProp := localRepoSystemProperty + localRepoDir
+
+	args := []string{"mvn", "clean", "install", repoLocalSystemProp}
+	args = append(args, additionalArgs...)
+	runCli(t, args...)
+	err := os.Chdir(oldHomeDir)
```

Comment:
This applies to other usages of ```os.Chdir``` in this file - 
 Let's move this into a defer func, which is set right after ```changeWD``` . The defer should also asset the error - 
```
	defer func() {
		err := os.Chdir(oldHomeDir)
		assert.NoError(t, err)
	}()
```